optuna_dashboard/ts/components/ModelViewer.tsx

```diff
@@ -0,0 +1,109 @@
+import * as THREE from "three"
+import React, { useEffect, useState } from "react"
+import { Canvas } from "@react-three/fiber"
+import { GizmoHelper, GizmoViewport, OrbitControls } from "@react-three/drei"
+import { STLLoader } from "three/examples/jsm/loaders/STLLoader"
+import { Rhino3dmLoader } from "three/examples/jsm/loaders/3DMLoader"
+import { PerspectiveCamera } from "three"
+
+interface ModelViewerProps {
+  src: string
+  width: string
+  height: string
+  hasGizmo: boolean
+  filetype: string | undefined
+}
+
+function CustomGizmoHelper(): JSX.Element {
+  return (
+    <GizmoHelper alignment="bottom-right" margin={[80, 80]}>
+      <GizmoViewport
+        axisColors={["red", "green", "skyblue"]}
+        labelColor="black"
+      />
+    </GizmoHelper>
+  )
+}
+
+const calculateBoundingBox = (geometries: THREE.BufferGeometry[]) => {
+  const boundingBox = new THREE.Box3()
+  geometries.forEach((geometry) => {
+    const mesh = new THREE.Mesh(geometry)
+    boundingBox.expandByObject(mesh)
+  })
+  return boundingBox
+}
+
+export function ModelViewer(props: ModelViewerProps): JSX.Element {
```

Comment:
I think the name `ModelViewer` is a bit ambiguous as it may be associated with machine learning "model".  How about naming it like below?

```suggestion
export const ThreejsArtifactViewer: React.FC<ModelViewerProps> = (props) => {
```